src/main/java/frc/robot/subsystems/AlignerSubsystem.java

```diff
@@ -0,0 +1,103 @@
+package frc.robot.subsystems;
+
+import static frc.robot.Constants.AlignerConstants.*;
+
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.motorcontrol.MotorUtil;
+import frc.robot.shuffleboard.GRTNetworkTableEntry;
+import frc.robot.shuffleboard.GRTShuffleboardTab;
+
+public class AlignerSubsystem extends SubsystemBase {
+    // current target position for motors
+    public double current_slaptarget = OPENSLAP;
+    public double current_anglertarget = OPENANGLER;
+    // current action state
+    public boolean to_slap;
+    public boolean to_grab;
+    // state of angler(open or closed)
+    public boolean angler_open = true;
+
+    private final WPI_TalonSRX motorSlapper = MotorUtil.createTalonSRX(SLAPID);
+    private final WPI_TalonSRX motorAngler = MotorUtil.createTalonSRX(ANGLEID);
+
+    private final GRTShuffleboardTab shuffleboardTab = new GRTShuffleboardTab("Aligner");
+    private final GRTNetworkTableEntry slapperPositionEntry = shuffleboardTab.addEntry("Slapperposition", 0).at(0, 0);
+    private final GRTNetworkTableEntry anglerPositionEntry = shuffleboardTab.addEntry("Anglerposition", 0).at(1, 0);
+
+    // variables for motor positions
+    public double slappos;
+    public double anglerpos;
+
+    public AlignerSubsystem() {
+        // set motors to brake
+        motorSlapper.setNeutralMode(NeutralMode.Brake);
+        motorAngler.setNeutralMode(NeutralMode.Brake);
+
+        // setup encoders for motors
+        motorSlapper.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+        motorSlapper.setSensorPhase(true);
+
+        motorAngler.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+        motorAngler.setSensorPhase(true);
+
+    }
+    // as of now:
+    // motor can potentially get stuck going back and forth depending on speed
+    // no code for catching/fixing motor over-turning
+
+    public void periodic() {
+
+        // get current motor positions
+        slappos = motorSlapper.getSelectedSensorPosition();
+        anglerpos = motorAngler.getSelectedSensorPosition();
+
+        // check if slapper trigger is pressed
+        if (to_slap) {
+            current_slaptarget = SLAPSLAP;
+        }
+        // check if grab trigger is pressed and angler is open
+        else if (to_grab && angler_open) {
+            current_slaptarget = CLOSEDSLAP;
+            current_anglertarget = CLOSEDANGLER;
+            angler_open = false;
+        }
+        // check if grab trigger is pressed and angler is closed
+        else if (to_grab && !angler_open) {
+            current_slaptarget = OPENSLAP;
+            current_anglertarget = OPENANGLER;
+            angler_open = true;
```

Comment:
wouldn't these two states just flip hella fast if the grab trigger is held down?

i also don't really get angler_open, how is it different from to_grab? I think it would make more sense to put toggling logic in RobotContainer so that all the states in this subsystem directly represent what we want the hardware to be doing, rather than like a button state (that we later process using toggle logic).